### PR TITLE
Refactor, update Kotlin to v1.9.22

### PR DESCRIPTION
**Pull request by PatrykMis**
- Update Kotlin to v1.9.22 and related dependencies
- migrate away from internal APIs
- set apk filename much easier way
- fix typo in variable
- remove javapoet as it's not being used in our intire project (just transitive dependency)
- get rid of plugin conflicts so proper plugins declarations can be used
- simplify CI workflow to just upload all generated APKs instead of hardcoding full paths and filenames

### CODE REVIEW

**Comment by PatrykMis** [comment]
@Irineu333, while refactoring, I discovered that you are using `com.android.build.gradle.internal.dsl.BaseAppModuleExtension`. If you migrate from an internal to a public API, we can simplify the code by removing `libs.android.gradlePlugin` and relying solely on AGP's API. Consider the following replacements:

* [ApplicationBaseFlavor](https://developer.android.com/reference/tools/gradle-api/8.2/com/android/build/api/dsl/ApplicationBaseFlavor)
* [ApplicationDefaultConfig](https://developer.android.com/reference/tools/gradle-api/8.2/com/android/build/api/dsl/ApplicationDefaultConfig)
* [ApplicationAndroidComponentsExtension](https://developer.android.com/reference/tools/gradle-api/8.2/com/android/build/api/variant/ApplicationAndroidComponentsExtension)

**Comment by PatrykMis** [comment]
@Irineu333, I finally managed to do it. Perhaps it's not an ideal solution, but for now, it's replaced by [ApplicationExtension](https://developer.android.com/reference/tools/gradle-api/8.2/com/android/build/api/dsl/ApplicationExtension).

File: app/build.gradle.kts [hunk 1]
```diff
@@ -19,16 +19,16 @@
 @file:Suppress("UnstableApiUsage")
 
 plugins {
-    id(libs.plugins.android.application.get().pluginId)
-    id(libs.plugins.kotlin.android.get().pluginId)
```
Comment (Irineu333) [inline app/build.gradle.kts:23]: Cool, finally fixed it?

File: gradle.properties [hunk 2]
```diff
@@ -10,7 +10,7 @@ org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
+org.gradle.parallel=true
```
Comment (Irineu333) [inline gradle.properties:13]: What does it do?

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -24,8 +23,6 @@ timber = { group = "com.jakewharton.timber", name = "timber", version.ref = "tim
 # Plugins
 android-gradlePlugin = { group = "com.android.tools.build", name = "gradle", version.ref = "agp" }
 android-gradlePluginApi = { group = "com.android.tools.build", name = "gradle-api", version.ref = "agp" }
-kotlin-gradlePlugin = { group = "org.jetbrains.kotlin", name = "kotlin-gradle-plugin", version.ref = "kotlin" }
-squareup-javapoet = { group = "com.squareup", name = "javapoet", version.ref = "javapoet" }
```
Comment (Irineu333) [inline gradle/libs.versions.toml:28]: I don't remember why I added javapoet anymore. I think it was some version conflict, which forced me to specify a version. It was added in PR #103.

File: buildSrc/src/main/kotlin/VersionConfig.kt [hunk 4]
```diff
@@ -72,6 +72,6 @@ data class VersionConfig(
     companion object {
         private const val MAJOR_WEIGHT = 100
         private const val MINOR_WEIGHT = 10
-        private const val PATH_WEIGHT = 1
+        private const val PATCH_WEIGHT = 1
```
Comment (Irineu333) [inline buildSrc/src/main/kotlin/VersionConfig.kt:75]: Thanks!

File: buildSrc/src/main/kotlin/Project.kt [hunk 5]
```diff
@@ -16,7 +16,7 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-import com.android.build.gradle.internal.dsl.BaseAppModuleExtension
+import com.android.build.api.dsl.ApplicationExtension
```
Comment (Irineu333) [inline buildSrc/src/main/kotlin/Project.kt:19]: Thank you for the improvement!

File: app/build.gradle.kts [hunk 6]
```diff
@@ -61,6 +61,8 @@ android {
     defaultConfig {
         applicationId = "com.neo.speaktouch"
 
+        configure<BasePluginExtension> { archivesName.set(rootProject.name) }
```
Comment (Irineu333) [inline app/build.gradle.kts:64]: Where did this code go?

File: app/build.gradle.kts (hunk 1 above)
Comment (PatrykMis) [reply app/build.gradle.kts:23]: Yes, removing `gradlePlugin` from `buildSrc` made able to declare it normally.

File: app/build.gradle.kts (hunk 6 above)
Comment (PatrykMis) [reply app/build.gradle.kts:64]: I'm not sure what you mean but it sets base apk name. Previously, would like to just add `setProp("archiveBaseName", rootProject.name)` but it's deprecated according to [this](https://docs.gradle.org/8.5/userguide/upgrading_version_8.html#base_convention_deprecation)

File: gradle.properties (hunk 2 above)
Comment (PatrykMis) [reply gradle.properties:13]: It enables parallel builds. In our case, the test app is not dependent on main app so they can build independantly.

File: gradle/libs.versions.toml (hunk 3 above)
Comment (PatrykMis) [reply gradle/libs.versions.toml:28]: It's possible, but now it is working fine especially after we sorted some things out.

**Review by Irineu333 — APPROVED** [review]

**Pull request merged**